lib/TPP/Transforms/TileConsumerAndFuseProducers.cpp

```diff
@@ -449,7 +449,32 @@ static FailureOr<scf::SCFTileAndFuseResult> fuseWithEltwise(
 // Trivial tile selection. If the dimension is statically known, it perfectly
 // divides the tile, and we have enough iterations return a default of 32.
 static int64_t getTileForDim(linalg::LinalgOp linalgOp, unsigned dim) {
-  const int64_t tile = 32;
+  int64_t tile = 32;
+
+  // Check if a tile size hint is associated to the IR via DLTI.
+  auto deriveFromDLTI = [&](ModuleOp moduleOp) {
```

Comment:
I'd move most of that to a helper where you pass device id, spec id and it returns an attribute. This should eventually be upstream in some query helper.

Reply:
Yes, that's what I would like as well as an API.The issue I was having is with communicating back why a lookup failed. A utility function like that has many failure modes. Should we have two util functions, or a flag, whereby one behaviour is to just return a `Optional<Attribute>`/`FailureOr<Attribute>` and the other one does that and emits messages for each distinct failure case.

Reply:
Fair enough, we can start this way for now and evolve into that.